MoMMI/Modules/github.py

```diff
@@ -765,9 +766,82 @@ def gh_register_help() -> None:
             async with session.post(post) as resp:
                 await resp.text()
 
+#todo
+# support short label codes like qol, bugfix etc, so not search for labels literally
+# filter for emojicracy, just use emoji-modifiers to calc total value of issue, then choose between top ~10ish
+#   /repos/:owner/:repo/issues/:issue_number/reactions?content=+1 (or hooray, heart, confused, laugh, -1)
+#   needs Accept: application/json in header
+# make it possible to harddefine params eg. repo = bla/bla, so you don't have to give a label to search other repos
+#   would also be nice to have when the emojicracy-filter gets to be a thing
+# dont use \w
+@command("giveissue", r"giveissue(?:\s+(-\w+=\w+(?:\s+-\w+=\w+)*))?")
+async def giveissue_command(channel: MChannel, match: Match, message: Message) -> None:
+    try:
+        cfg: List[Dict[str, Any]] = channel.server_config("modules.github.repos")
+    except:
+        # Server has no config settings for GitHub.
+        await channel.send(":trash: No config found")
+        return
+
+    await channel.send(":hourglass_flowing_sand: Fetching random issue")
+
+    #default params
+    #repo = "vgstation-coders/vgstation13"
+    prefix = ""
+    labels = ""
+    emote = ""
+    ranking_limit = 20
+
+    
+    #getting params
+    text_params = [x.strip() for x in match.group(1).split(" -")] # strip whitespaces
+    
+    for param in text_params:
+        temp = param.split("=")
+        if temp[0] == "prefix":
+            prefix = temp[1].strip()
+            continue
+        if temp[0] == "labels":
+            labels = temp[1].strip()
+            continue
+        if temp[0] == "emote":
+            emote = re.search(REG_GIT_EMOTE, temp[1]).group(0)
+            continue
+        if temp[0] == "limit":
+            ranking_limit = int(temp[1])
+            continue
+        await channel.send(f":trash: Warning: Unknown parameter: {temp[0]}")
+
+    for repo_config in cfg:
+        repo = repo_config["repo"]
+
+        if not is_repo_valid_for_command(repo_config, channel, prefix):
+            continue
+
+        issues = []
+        i = 1
+        while 1:
+            url = github_url(f"/repos/{repo}/issues")
+            part_issues = await get_github_object(url, {"labels" : labels, "page" : i, "per_page" : "100"})
```

Comment:
This doesn't filter out closed issues, does it?

Also, I would really like to avoid doing 25+ requests to GitHub for one command. For reference, you get about 5000 requests per hour with their API. Of course the rate limit does not kick in multiple times due to caching inside MoMMI, but it's still quite inefficient.

What you might be able to do is pull the first page, then pull the *last* page to get the total amount of issues, pick a number and then pick an issue. 3 requests per. You could also simply pick a random page and then pick among that page, 2 requests per, although that will give a slight bias to the last page.

Reply:
how would i get the last page? i read about `rel=last` but never got it to work

Reply:
GitHub returns the page count in a response header, you should be able to use that.